automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/orch/OrchestrationApiSpec.scala

```diff
@@ -44,14 +43,22 @@ class OrchestrationApiSpec extends FreeSpec with Matchers with ScalaFutures with
 
       withBillingProject("auto-goog-role") { projectName =>
         val preRoleFailure = googleBigQueryDAO.startQuery(userToken.value, GoogleProject(projectName), "meh").failed.futureValue
+
         preRoleFailure shouldBe a[GoogleJsonResponseException]
         preRoleFailure.getMessage should include(user.email)
         preRoleFailure.getMessage should include(projectName)
         preRoleFailure.getMessage should include("bigquery.jobs.create")
 
         Orchestration.billing.addGoogleRoleToBillingProjectUser(projectName, user.email, role)(ownerToken)
 
-        val queryReference = googleBigQueryDAO.startQuery(userToken.value, GoogleProject(projectName), shakespeareQuery).futureValue
+        // The google role might not have been applied the first time we call startQuery() - poll until it has
+        val queryReference = eventually {
+          val start = googleBigQueryDAO.startQuery(userToken.value, GoogleProject(projectName), shakespeareQuery).futureValue
+          start shouldBe a[JobReference]
```

Comment:
I think I understand this case... you're retrying the thing until the thing returns the type of result you expect if the thing was successful.